File: solutions/c/binary/1/src/binary.c
#include "binary.h"

int convert(const char *input) {
  int i = 0;
  int result = 0;
  for (i = 0; input[i] != '\0'; i++) {
    if (input[i] > '1')
      return INVALID;
    result <<= 1;
    if (input[i] == '1') {
      result |= 1;
    }
  }
  return result;
}

File: solutions/lua/binary/1/binary.lua
local function to_decimal(input)
    local result = 0;
    for i = 0, #input - 1 do
        tmp = input:byte(#input - i) - 48
        if tmp < 0 or tmp > 9 then
          return 0
        end
        result = result + (tmp * math.pow(2, i))
    end
    return result
end

return {
    to_decimal = to_decimal
}
